client/client.go

```diff
@@ -222,24 +223,28 @@ func (c *httpClusterClient) Do(ctx context.Context, act httpAction) (*http.Respo
 
 	var resp *http.Response
 	var body []byte
+	var cerr ClusterError
 	var err error
 
 	for _, ep := range eps {
 		hc := c.clientFactory(ep)
 		resp, body, err = hc.Do(ctx, action)
 		if err != nil {
+			cerr.Errors = append(cerr.Errors, err)
 			if err == context.DeadlineExceeded || err == context.Canceled {
-				return nil, nil, err
+				return nil, nil, cerr
```

Comment:
Note that it is almost impossible for the client to know if it is a cluster side error or a client side error or a proxy error with a single error description.
Cluster error always means that cluster is not available in the view of the client with some given condition.

That is why we provide a cluster error that contains all encountered errors, so that user can assert the error type and see the detailed errors.